packages/eslint-plugin/tsconfig.json

```diff
@@ -3,8 +3,8 @@
     "noEmit": true,
     "allowJs": true,
     "checkJs": true,
-    "module": "commonjs",
-    "moduleResolution": "nodenext",
+    "module": "NodeNext",
```

Comment:
related to: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-2.html#module-and-moduleresolution-must-match-under-recent-nodejs-settings